sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts

```diff
@@ -108,10 +113,38 @@ export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
         "Not all required fields to update a received MSFAA record were provided.",
       );
     }
-
+    // Check if the msfaa number needs to be reactivated or simply updated with a signed date.
+    const retrievedMSFAARecord = await this.repo.findOne({
+      select: {
+        id: true,
+        studentId: true,
+        referenceApplicationId: true,
+        cancelledDate: true,
+        dateSigned: true,
+        offeringIntensity: true,
+      },
+      where: {
+        msfaaNumber,
+      },
+    });
+    if (
```

Comment:
Please re-validate the condition which identifies previously cancelled MSFAA. As per the current logic we do not set datesigned to null while cancellling MSFAA.



![image](https://github.com/bcgov/SIMS/assets/54600590/cfa51fe1-8241-4e2f-930f-f465687104c1)